modules/checklist/steps-manager.php

```diff
@@ -71,13 +77,7 @@ public function mark_step_as_completed( string $step_id ) : void {
 	 * @return void
 	 */
 	public function unmark_step_as_completed( string $step_id ) : void {
-		foreach ( $this->step_instances as $step ) {
-			if ( $step->get_id() === $step_id ) {
-				$step->unmark_as_completed();
-
-				return;
-			}
-		}
+		$this->update_step( $step_id, [ Step_Base::MARKED_AS_COMPLETED_KEY => false ] );
```

Comment:
I wonder, should we have one function called `set_step_state` and pass the desire state to it?
E.g, 
`set_step_state( string $step_id, bool true );`
`set_step_state( string $step_id, bool false );`

WDYT?